design-at-cornell/src/clubs/dashboard/ClubBubble.tsx

```diff
@@ -8,39 +8,89 @@ import {
-    onClick={() => (window.location.href = club.content.website)}
-  >
-    <Subtitle>
-      <p>{club.content.orgType}</p>
-      <img src={bookmark} alt="save club" />
-    </Subtitle>
-    <Title>{club.title}</Title>
-    <TagsContainer>
-      {club.content.size === '' ? null : (
-        <Tag
-          style={{
-            background: dashboardColors[Math.floor(Math.random() * dashboardColors.length)],
-          }}
-        >
-          {club.content.size}
-        </Tag>
-      )}
-      {club.content.designAreas.filter((area) => area !== '').length === 0 ? null : (
-        <Tag
-          style={{
-            background: dashboardColors[Math.floor(Math.random() * dashboardColors.length)],
-          }}
-        >
-          {club.content.designAreas.join(', ')}
-        </Tag>
-      )}
-    </TagsContainer>
-  </ElementContainer>
-);
+const ClubBubble = (club: Club) => {
+  const [open, setOpen] = React.useState(false);
+
+  const clubCard = (
+    <ElementContainer
+      style={{ borderColor: dashboardColors[Math.floor(Math.random() * dashboardColors.length)] }}
+    >
+      <Subtitle>
+        <p>{club.content.orgType}</p>
+        <img src={bookmark} alt="save club" />
+      </Subtitle>
+      <Title>{club.title}</Title>
+      <TagsContainer>
+        {club.content.size === '' ? null : (
+          <Tag
+            style={{
+              background: dashboardColors[Math.floor(Math.random() * dashboardColors.length)],
+            }}
+          >
+            {club.content.size}
+          </Tag>
+        )}
+        {club.content.designAreas.filter((area) => area !== '').length === 0 ? null : (
+          <Tag
+            style={{
+              background: dashboardColors[Math.floor(Math.random() * dashboardColors.length)],
+            }}
+          >
+            {club.content.designAreas.join(', ')}
+          </Tag>
+        )}
+      </TagsContainer>
+    </ElementContainer>
+  );
+
+  return (
+    <Modal
+      onClose={() => setOpen(false)}
+      onOpen={() => setOpen(true)}
+      open={open}
+      trigger={clubCard}
+      style={{ width: '790px', height: '527px', borderRadius: '10px' }}
+    >
+      <ModalContainer>
+        <VerticalModalHeader>
+          <Title >{club.title}</Title>
+          <Subtitle>
+            <p style={{ color: 'white' }}>{'Apply by '}</p>
+          </Subtitle>
+          <TagsContainer>
+            <Tag>{club.content.orgType}</Tag>
+            <Tag>{club.content.size}</Tag>
+            {club.content.designAreas.filter((x) => x !== '').length === 0 ? null : (
+              <Tag>{club.content.designAreas.join(', ')}</Tag>
+            )}
+            <Tag>{club.content.credits + ' Credits'}</Tag>
+          </TagsContainer>
+          <ButtonContainer>
+            <RectangularButton>
+              <img onClick={() => (window.location.href = club.content.website)} src={bookmark} alt={'website'} />
```

Comment:
the `onClick` method should be added to the button component instead of `img` (right now it only redirects to the website when we click the icon, but we want it to happen when we click anywhere on the button)